src/yunohost/backup.py

```diff
@@ -1232,46 +1231,35 @@ def _restore_system(self):
 
         regen_conf()
 
-        # Check if we need to do the migration 0009 : setup group and permission
+        # Check that at least a group exists (all_users) to know if we need to
+        # do the migration 0011 : setup group and permission
+        #
         # Legacy code
-        result = ldap.search('ou=groups,dc=yunohost,dc=org',
-                             '(&(objectclass=groupOfNamesYnh)(cn=all_users))',
-                             ['cn'])
-        if not result:
+        if not "all_users" in user_group_list()["groups"].keys():
             from yunohost.tools import _get_migration_by_name
             setup_group_permission = _get_migration_by_name("setup_group_permission")
             # Update LDAP schema restart slapd
             logger.info(m18n.n("migration_0011_update_LDAP_schema"))
             regen_conf(names=['slapd'], force=True)
             setup_group_permission.migrate_LDAP_db()
 
-        # Remove all permission for all app which sill in the LDAP
-        for per in ldap.search('ou=permission,dc=yunohost,dc=org',
-                               '(&(objectClass=permissionYnh)(!(cn=main.mail))(!(cn=main.metronome))(!(cn=main.sftp)))',
-                               ['cn']):
-            if not ldap.remove('cn=%s,ou=permission' % per['cn'][0]):
-                raise YunohostError('permission_deletion_failed',
-                                    permission=per['cn'][0].split('.')[0],
-                                    app=per['cn'][0].split('.')[1])
+        # Remove all permission for all app which is still in the LDAP
+        for permission_name in user_permission_list(ignore_system_perms=True)["permissions"].keys():
+            permission_delete(permission_name, force=True)
 
         # Restore permission for the app which is installed
-        for per in old_apps_permission:
-            try:
-                permission_name, app_name = per['cn'][0].split('.')
-            except:
-                logger.warning(m18n.n('permission_name_not_valid', permission=per['cn'][0]))
+        for permission_name, permission_infos in old_apps_permission.items():
+            app_name = permission_name.split(".")[0]
             if _is_installed(app_name):
-                if not ldap.add('cn=%s,ou=permission' % per['cn'][0], per):
-                    raise YunohostError('apps_permission_restoration_failed', permission=permission_name, app=app_name)
-
+                permission_create(permission_name, urls=permission_infos["urls"], sync_perm=False)
+                user_permission_update(permission_name, remove="all_users", add=permission_infos["allowed"])
```

Comment:
In some case we really use a edge case for this. By example if the allowed group was `all_users`, we are in a state we the arg `remove` and `add` contains `all_users`. So is this a really good solution ?

Reply:
Ah yes good catch ! We probably need to have an `if permissions["allowed"] != default_permissions` or something like this